x-pack/legacy/plugins/file_upload/public/components/json_index_file_picker.js

```diff
@@ -39,60 +39,85 @@ export class JsonIndexFilePicker extends Component {
+          defaultMessage: 'File size error: {errorMessage}',
           values: {
             errorMessage: error.message
           }
         })
       });
       return;
     }
+    return fileSizeValid;
+  }
 
-    this.props.setIndexName(initIndexName);
-    this._parseFile(file);
-  };
-
-  _getIndexName({ name, size }) {
-    if (!name) {
-      throw new Error(i18n.translate('xpack.fileUpload.jsonIndexFilePicker.noFileNameError', {
-        defaultMessage: 'No file name provided'
-      }));
-    }
+  _getFileNameAndCheckType({ name }) {
+    let fileNameOnly;
+    try {
+      if (!name) {
+        throw new Error(i18n.translate('xpack.fileUpload.jsonIndexFilePicker.noFileNameError', {
+          defaultMessage: 'No file name provided'
+        }));
+      }
 
-    const splitNameArr = name.split('.');
-    const fileType = splitNameArr.pop();
-    if (!ACCEPTABLE_FILETYPES.includes(fileType)) {
-      throw new Error(i18n.translate('xpack.fileUpload.jsonIndexFilePicker.acceptableTypesError', {
-        defaultMessage: 'File is not one of acceptable types: {types}',
-        values: {
-          types: ACCEPTABLE_FILETYPES.join(', ')
-        }
-      }));
-    }
+      const splitNameArr = name.split('.');
+      const fileType = splitNameArr.pop();
+      if (!ACCEPTABLE_FILETYPES.includes(fileType)) {
+        throw new Error(i18n.translate('xpack.fileUpload.jsonIndexFilePicker.acceptableTypesError', {
+          defaultMessage: 'File is not one of acceptable types: {types}',
+          values: {
+            types: ACCEPTABLE_FILETYPES.join(', ')
+          }
+        }));
+      }
 
-    if (size > MAX_FILE_SIZE) {
-      throw new Error(i18n.translate('xpack.fileUpload.jsonIndexFilePicker.acceptableFileSize', {
-        defaultMessage: 'File size {fileSize} bytes exceeds max file size of {maxFileSize}',
-        values: {
-          fileSize: size,
-          maxFileSize: MAX_FILE_SIZE
-        }
-      }));
+      fileNameOnly = splitNameArr[0];
+    } catch (error) {
+      this.setState({
+        fileUploadError: i18n.translate('xpack.fileUpload.jsonIndexFilePicker.fileProcessingError', {
+          defaultMessage: 'File processing error: {errorMessage}',
+          values: {
+            errorMessage: error.message
+          }
+        })
+      });
+      return;
     }
-
-    return splitNameArr[0];
+    return fileNameOnly.toLowerCase();
   }
 
   async _parseFile(file) {
     const {
-      setFileRef, setParsedFile, resetFileAndIndexSettings, onFileUpload, transformDetails
+      setFileRef, setParsedFile, resetFileAndIndexSettings, onFileUpload,
+      transformDetails, setIndexName
     } = this.props;
+
+    const fileSizeValid = this._checkFileSize(file);
+    const defaultIndexName = this._getFileNameAndCheckType(file);
+    if (!fileSizeValid || !defaultIndexName) {
+      resetFileAndIndexSettings();
+      return;
+    }
```

Comment:
A nit, but I wonder if we could also validate the file against the GeoJSON schema before we try to pass it to jsts? If you upload a JSON file that is not GeoJSON it returns an ambiguous error `Unable to parse file: Unknown GeoJSON type: undefined`.

Reply:
This fix/improvement doesn't fit in this particular issue, but I definitely agree that this would be an improvement. I've captured it in #47472